lib/readyset.rb

```diff
@@ -97,13 +98,16 @@ def self.raw_query(*sql_array) # :nodoc:
   # @yield a block whose queries should be routed to ReadySet
   # @return the value of the last line of the block
   def self.route(prevent_writes: true, &block)
+    QueryAnnotator.routing_to_readyset = true
     if prevent_writes
       ActiveRecord::Base.
         connected_to(role: reading_role, shard: shard, prevent_writes: true, &block)
     else
       ActiveRecord::Base.
         connected_to(role: writing_role, shard: shard, prevent_writes: false, &block)
     end
+  ensure
+    QueryAnnotator.routing_to_readyset = false
```

Comment:
Can we add some test coverage to check that `Readyset.route` sets `routing_to_readyset` to true at the start of the method and then always sets it to false at the end of the method (even in the case of an error)?

Reply:
Added!